iOS/DuckDuckGo/DaxEasterEggLogos/DaxEasterEggHandler.swift

```diff
@@ -0,0 +1,170 @@
+    /// Should only be called on DuckDuckGo search pages.
+    func extractLogosForCurrentPage()
+    
+    /// Processes a raw logo URL extracted by JavaScript.
+    /// Converts relative paths to absolute URLs and handles the "themed|" prefix format.
+    ///
+    /// - Parameters:
+    ///   - logoURL: Raw logo URL from JavaScript (format: "themed|/path")
+    ///   - pageURL: URL of the page where logo was extracted
+    func didExtractLogo(_ logoURL: String?, from pageURL: String)
+    
+}
+
+/// Handler that manages extraction and processing of dynamic logos from DuckDuckGo search pages.
+///
+/// This class executes JavaScript directly to find logos and processes the results for the native UI.
+/// It processes raw logo URLs by converting relative paths to absolute URLs and handling 
+/// DuckDuckGo's "themed|" URL format.
+///
+/// The handler is created on-demand when navigating to DuckDuckGo search pages.
+public class DaxEasterEggHandler: DaxEasterEggHandling {
+    
+    public weak var delegate: DaxEasterEggDelegate?
+    private weak var webView: WKWebView?
+    
+    public init(webView: WKWebView) {
+        self.webView = webView
+    }
+    
+    public func extractLogosForCurrentPage() {
+        guard let webView = webView else {
+            Logger.daxEasterEgg.debug("extractLogosForCurrentPage - webView is nil")
+            return
+        }
+        
+        Logger.daxEasterEgg.debug("extractLogosForCurrentPage - executing JavaScript directly on URL: \(webView.url?.absoluteString ?? "no-url")")
+        
+        // Execute logo extraction JavaScript directly without UserScript dependency
+        executeLogoExtraction(webView: webView)
+    }
+    
+    private func executeLogoExtraction(webView: WKWebView) {
+        let extractionJS = """
+        (function() {
+            try {
+                function findLogo() {
+                    var ddgLogo = document.querySelector('.js-logo-ddg');
+                    
+                    if (!ddgLogo) {
+                        ddgLogo = document.querySelector('.logo-dynamic');
+                    }
+                    if (!ddgLogo) {
+                        ddgLogo = document.querySelector('[data-dynamic-logo]');
+                    }
+                    
+                    if (!ddgLogo) {
+                        return null;
+                    }
+                    
+                    if (ddgLogo.dataset && ddgLogo.dataset.dynamicLogo) {
+                        return 'themed|' + ddgLogo.dataset.dynamicLogo;
+                    }
+                    
+                    return null;
+                }
+                
+                return findLogo();
+            } catch (error) {
+                console.error('DaxEasterEgg: Error in logo extraction:', error);
+                return null;
+            }
+        })();
+        """
+        
+        webView.evaluateJavaScript(extractionJS) { [weak self] result, error in
```

Comment:
Why this rather than async/await?

Reply:
Thanks @brindy. No reason, just habit. Converted to async/await now.